content/movie-review-pt2.md

```diff
@@ -0,0 +1,520 @@
+# Movie Review Program Pt2
+
+# Lesson Objectives
+
+- Describe the process of creating a new account using a Program Derived Address (PDA)
+- Use seeds to derive a PDA
+- Use the space required by an account to calculate the amount of rent (in lamports) a user must allocate
+- Use a Cross Program Invocation (CPI) to initialize an account with a PDA as the address of the new account
+- Explain how to update the data stored on a new account
+
+# TL;DR
+
+- Creating an account requires that we calculate the space and rent to allocate for a new account
+- A Program Derived Address (PDA) is derived from a program ID and an optional list of seeds
+- Creating a new account requires that perform a Cross Program Invocation (CPI) to the `create_account` instruction on the System Program
+- Updating the data field on an account requires that we serialize (convert to byte array) the data into the account
+
+# Overview
+
+In the previous lesson we went over how to, deserialize instruction data, deserialize account data, and print messages to the program log.
+
+In this lesson we will build upon what we have already done. The next step is to create a new account to store the movie review data from the user.
+
+## Account State
+
+In order to store data on a new account, we must first define a struct that serves as the format for the data field of the account. These structs are stored in a `state.rs` file and are used to serialize data into an account and deserialize data from an existing account.
+
+For our movie review program, we will store the rating, description, and title provided by the user. We will also want a field to check whether the account storing the review has already been initialized. This is to prevent duplicate reviews of the same movie by the same user. We’ll refer to this format as the `MovieAccountState`.
+
+To store data on an account, we must serialize the data. Serialization is the process of converting an object into a bytes array (an area of memory containing a continuous sequence of bytes). For our use case, the object is simply the data we want to store. Writing to the data field of an account requires that we first serialization the data we want to store.
```

Comment:
Let's break this section into two ### subsections that are part of the same ## section. The first subsection is like "Represent state as a Rust type" or something and the next is something like "Prepare type for Borsh". I don't know exactly the titles - you can play with it. But I think it helps to be super clear.

That being said, I think we can do that with the entire lesson - break it into main headings with subsections. Probably something like:
* Use account data for state management
  * Represent state as a Rust type
  * Prepare type for Borsh
* Create state accounts
  * Derive the account address
  * Space and rent
  * Use System Program to create account
* Read from and write to state accounts
  * Deserialize state account data
  * Serialize state account data

Again, I don't know if those are exactly the right titles, but that seems to be the flow, right? Lmk what you think

Reply:
Once we've settled on those headings/flow/etc, you may need to tweak some of the language at the beginning of each section/subsection to match

Reply:
Just going to tag both of you @ZYJLiu and @mixelpixel do you have thoughts on the flow?

Reply:
I like that breakdown - there's a bunch of big content coming together in this Overview - the sub-headings help a lot 👍